app/models/ticket.rb

```diff
@@ -1,4 +1,9 @@
 class Ticket < ApplicationRecord
   belongs_to :reservation
   validates :seat, presence: true
+  validate :seat_available?
+
+  def seat_available?
+    raise StandardError, 'Seat taken' unless Reservation.find(reservation_id).screening.available_seats.include? seat
```

Comment:
Also you can refer to tickets reservation directly, active record will find it for you
```suggestion
    errors.add(seat: 'Seat taken') unless reservation.screening.available_seats.include? seat
```